plugins/qc-firehose/README.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Plugin: Qc Firehose
+---
+
+## Introduction
+
+Qualcomm MSM based devices contain a special mode of operation, called Emergency Download Mode.
+In this mode, the device identifies itself as Qualcomm HS-USB 9008 through USB, and can communicate
+with a PC host. EDL is implemented by the SoC ROM code (also called PBL).
+
+The EDL mode itself implements the Qualcomm Sahara protocol, which accepts an OEM-digitally-signed
+payload over USB.
+
+## Firmware Format
+
+The daemon will decompress the cabinet archive and then extract a zip payload. Inside this payload
+there must be:
+
+* One saraha binary with the `.mbn` extension that is suitable for the target platform
```

Comment:
This is the Firehose programmer/loader binary, which may have the `.mbn` or `.elf` extension. For the modem-manager plugin we simply always call it  `firehose-prog.mbn`.

Reply:
Do you think it makes sense to search for .elf as well?

Reply:
Although maybe we could remove [ModemManagerFirehoseProgFile](https://github.com/fwupd/fwupd/blob/1a3d9141bca2547c0064fa0e5404539817695f1a/plugins/modem-manager/modem-manager.quirk#L134) if every Firehose loader keeps its original name.

Reply:
I think so too!